### PR TITLE
This PR adds a new fact puppet_config

### PR DESCRIPTION
**Pull request by b4ldr**
#### Pull Request (PR) description
This new fact aims to expose some of the configurations parameters of
puppet.  it is similar to the puppet_settings fact in stdlib but uses
structured facts and exposes a few more settings.

I first attempted to submit something similar to stdlib[1] but got
redirected here.  happy to update if this is useful or abandon if not

[1]https://github.com/puppetlabs/puppetlabs-stdlib/pull/1063

### CODE REVIEW

File: spec/unit/facter/puppet_settings.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+# not sure how best to predict this? the following won't work on windows
+hostname = `hostname -f`.strip
+settings = {
+  "main" => {
+    "hostcert" => "/dev/null/ssl/certs/#{hostname}.pem",
```
Comment (ekohl) [inline spec/unit/facter/puppet_settings.rb]: Technically Puppet uses the `certname` setting and not hostname. That just defaults to the hostname.

File: spec/unit/facter/puppet_settings.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "ssldir" => '/dev/null/ssl',
+    "vardir" => '/dev/null',
+  },
+  "master" => {
+    "localcacert" => '/dev/null/ssl/certs/ca.pem',
+    "ssldir" => '/dev/null/ssl',
+  }
+}
+describe 'puppet_config' do
+  let(:node) { 'puppet_config.example.org' }
```
Comment (ekohl) [inline spec/unit/facter/puppet_settings.rb]: Underscores in hostnames are not allowed

Comment (b4ldr) [reply spec/unit/facter/puppet_settings.rb]: This didn;t actully work and should have been removed, thanks

**Comment by b4ldr** [comment]
> Looks like Rubocop is still unhappy about the keys

oh sorry forgot to check that, should be sorted now

**Comment by ekohl** [comment]
Aren't these also available as `$settngs::setting`? https://puppet.com/docs/puppet/latest/lang_facts_builtin_variables.html suggests they are.

**Comment by ekohl** [comment]
Aren't these also available as `$settngs::setting`? https://puppet.com/docs/puppet/latest/lang_facts_builtin_variables.html suggests they are.

**Comment by b4ldr** [comment]
> Aren't these also available as `$settngs::setting`? https://puppet.com/docs/puppet/latest/lang_facts_builtin_variables.html suggests they are.

`$settngs::setting` resolves on the compile master, theses give the settings of the host

**Comment by bastelfreak** [comment]
@b4ldr thanks for the PR. Can you please have a look at the used email address in your commits? It isn't associated with your GitHub account. Please also document the fact in the README.md and maybe add some example output.

**Comment by b4ldr** [comment]
updated there dont seem to be any documented facts in either stdlib or here so i took a guess but of course happy to change

File: lib/facter/puppet_config.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    hostprivkey
+    hostcert
+    localcacert
+    ssldir
+    vardir
+    server
+  ]
+}
+Facter.add(:puppet_config) do
+  confine kernel: 'Linux'
```
Comment (vchepkov) [inline lib/facter/puppet_config.rb]: Why confine it for Linux only? It seems generic for other platforms as well?

Comment (b4ldr) [reply lib/facter/puppet_config.rb]: Thanks, i have created another #151  to remove this

**Pull request merged**